gsrs-module-substances-core/src/main/java/gsrs/module/substance/tasks/NameStandardizerTaskInitializer.java

```diff
@@ -0,0 +1,201 @@
+package gsrs.module.substance.tasks;
+
+import gov.nih.ncats.common.executors.BlockingSubmitExecutor;
+import gsrs.module.substance.repository.NameRepository;
+import gsrs.module.substance.repository.SubstanceRepository;
+import gsrs.module.substance.utils.NameStandardizer;
+import gsrs.scheduledTasks.ScheduledTaskInitializer;
+import gsrs.scheduledTasks.SchedulerPlugin;
+import gsrs.security.AdminService;
+import gsrs.springUtils.StaticContextAccessor;
+import ix.core.util.EntityUtils;
+import ix.ginas.models.v1.Name;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.TransactionDefinition;
+import org.springframework.transaction.support.TransactionTemplate;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Slf4j
+public class NameStandardizerTaskInitializer extends ScheduledTaskInitializer {
+
+    private String nameStandardizerClassName = null;
+    private String regenerateNameValue = "";
+
+    @Autowired
+    private SubstanceRepository substanceRepository;
+
+    @Autowired
+    private NameRepository nameRepository;
+
+    @Autowired
+    private AdminService adminService;
+
+    @Autowired
+    private PlatformTransactionManager platformTransactionManager;
+
+    @Override
+    public void run(SchedulerPlugin.JobStats stats, SchedulerPlugin.TaskListener l) {
+        if( this.nameStandardizerClassName== null || this.nameStandardizerClassName.length()==0) {
```

Comment:
This block should just use the fullStandardizer rather than trying to instantiate one here. I think it's a bit cleaner and closer to the validation rule.